src/SavedSearch.php

```diff
@@ -1195,6 +1214,31 @@ public function setEntityRecur(array $ids, $eid, $recur)
         return $result;
     }
 
+    /**
+     * Set visibility from massive actions
+     *
+     * @param array   $ids   Items IDs
+     * @param bool $is_private   Visibility
+     *
+     * @return boolean
+     */
+    public function setVisibility(array $ids, bool $is_private)
+    {
+        /** @var \DBmysql $DB */
+        global $DB;
+
+        $result = $DB->update(
+            $this->getTable(),
+            [
+                'is_private'  => $is_private,
+            ],
+            [
+                'id' => $ids
+            ]
+        );
```

Comment:
Updates should be done with the `CommonDBTM::update()` method. This is mandatory to permit plugins to use the update hooks.